extensions/gc/tasks/GarbageCollectorTask.js

```diff
@@ -254,7 +267,13 @@ class GarbageCollectorTask extends BackbeatTask {
                     .setUserMetadata({
                         'x-amz-meta-scal-s3-transition-attempt': undefined,
                     });
-                return this._putMetadata(entry, objMD, log, err => {
+                return this.retry({
+                    actionDesc: 'put metadata',
+                    logFields: { bucket, key, version },
+                    actionFunc: cb => this._putMetadata(entry, objMD, log, cb),
```

Comment:
Should probably set maxRetries on all retries

Reply:
Max retries are set to 2 ( 3 calls will be occurring with the first one into account )

Reply:
have you been able to understand why the default value is not used ?